src/context/context.ts

```diff
@@ -0,0 +1,13 @@
+import { createContext, Dispatch, SetStateAction } from 'react';
+import { Todo } from '../types/Todo';
+
+interface ContextProps {
+  todoLoadingStates: { [key: number]: boolean };
+  setTodoLoading: (id: number, status: boolean) => void;
+  setErrorMessage: (message: string) => void;
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+}
+
+export const TodoContext = createContext<ContextProps | undefined>(undefined);
```

Comment:
You can create a custom hook for your context so you don't need to pass it in `useContext` and it always will be defined, something like this:
```typescript
const useTodosContext = () => {
  const context = useContext(TodoContext);

  if (!context) {
    throw new Error('You can use context only inside the provider');
  }

  return context;
};
```